golang/lib/networks/network_context.go

```diff
@@ -202,75 +195,44 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
 		serviceId,
 		serviceIpAddr,
 		networkCtx.enclaveDataVolMountpoint,
-		containerCreationConfig.GetKurtosisVolumeMountpoint())
+		kurtosisVolumeMountpoint,
+		)
 	logrus.Tracef("New service successfully registered with Kurtosis API")
 
-	logrus.Trace("Loading static files into new service namespace...")
-	usedStaticFiles := containerCreationConfig.GetUsedStaticFiles()
-	staticFileAbsFilepathsOnService, err := serviceContext.LoadStaticFiles(usedStaticFiles)
-	if err != nil {
-		return nil, nil, stacktrace.Propagate(err, "An error occurred loading the following static files to service '%v': %+v", serviceId, usedStaticFiles)
-	}
-	logrus.Trace("Successfully loaded static files")
+	return serviceContext,  nil
+}
 
-	logrus.Trace("Initializing generated files...")
-	filesToGenerate := map[string]bool{}
-	for fileId := range containerCreationConfig.GetFileGeneratingFuncs() {
-		filesToGenerate[fileId] = true
-	}
-	generatedFileFilepaths, err := serviceContext.GenerateFiles(filesToGenerate)
-	if err != nil {
-		return nil, nil, stacktrace.Propagate(err, "An error occurred generating the files needed for service startup")
-	}
-	generatedFileAbsFilepathsOnService := map[string]string{}
-	for fileId, initializingFunc := range containerCreationConfig.GetFileGeneratingFuncs() {
-		filepaths, found := generatedFileFilepaths[fileId]
-		if !found {
-			return nil, nil, stacktrace.NewError(
-				"Needed to initialize file for file ID '%v', but no generated file filepaths were found for that file ID; this is a Kurtosis bug",
-				fileId)
-		}
-		fp, err := os.Create(filepaths.GetAbsoluteFilepathHere())
-		if err != nil {
-			return nil, nil, stacktrace.Propagate(err, "An error occurred opening file pointer for file '%v'", fileId)
-		}
-		if err := initializingFunc(fp); err != nil {
-			return nil, nil, stacktrace.Propagate(err, "The function to initialize file with ID '%v' returned an error", fileId)
-		}
-		generatedFileAbsFilepathsOnService[fileId] = filepaths.GetAbsoluteFilepathOnServiceContainer()
-	}
-	logrus.Trace("Successfully initialized generated files in suite execution volume")
+func (networkCtx *NetworkContext) StartService(
```

Comment:
I just had an insane thought, and LMK if this is a stupid idea: having Register and Start split as separate methods adds extra complexity. What if we use the same "do arbitrary filesystem things", but we ONLY take in a function? Something like:
```
func (networkCtx *NetworkContext) StartService(
    serviceId string,
    kurtosisVolumeMountpoint string,
    containerConfigSupplier func(ipAddr string, filesystem ServiceFilesystem) ContainerConfig
) {
```
?

Maybe that's a terrible idea, I dunno.... was just thinking that as soon as you have the register/start split, the user then needs to think about whether a service is registered-but-not-started, and whether their `ExecCommand` will work, we have to think about whether the `client` object inside `ServiceCtx` should be initialized already or not, we then have the ambiguous question of "where should StartService live? on the ServiceCtx or the NetworkCtx?"

Reply:
OOOooooo and actually, if we go that way, we wouldn't even need the `kurtosisVolumeMountpoint` because it'd already be specified in the `containerConfig` so we could just use that when creating the `ServiceCtx`!

Reply:
I like the idea of using only one method, now with your description, I understand why returning the `serviceContext` in RegisterService() wasn't a good idea, you are right it is confusing for users to know when they can call the Command and when the service is ready to be used.
The only thing that I'm not super happy about is passing the `containerConfigSupplier` because we will continue having this extra complexity for users in one of the first methods that they will be using in a Kurtosis Testsuite. 
if we do this, Wouldn't we still have these problems?
- user can't set the file extension
- user can't control the filename
- no way to create directories
- no way to put static DIRECTORIES (rather than files)
- no way to read the service container's files

Let's talk about it because maybe I'm not understanding everything